packages/stark-ui/src/modules/date-picker/components/date-picker.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, EventEmitter, ViewEncapsulation, Input, Output, ViewChild } from "@angular/core";
+import { MatDatepicker, MatDatepickerInput, MatDatepickerInputEvent } from "@angular/material";
+import * as moment from "moment";
+
+export type StarkDatePickerFilter = "OnlyWeekends" | "OnlyWeekdays" | ((date: Date) => boolean) | undefined;
+
+/**
+ * Component to display the stark date-picker
+ */
+@Component({
+	selector: "stark-date-picker",
+	templateUrl: "./date-picker.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkDatePickerComponent {
+	/**
+	 * Source Date to be bound to the datepicker model
+	 */
+	@Input()
+	public date: Date;
+
+	/**
+	 * Filter function or a string
+	 * Whenever this value is changed, we set the dateFilterFunc
+	 */
+	@Input()
+	public get dateFilter(): "OnlyWeekends" | "OnlyWeekdays" | ((date: Date) => boolean) | undefined {
+		return this._dateFilter;
+	}
+	public set dateFilter(value: "OnlyWeekends" | "OnlyWeekdays" | ((date: Date) => boolean) | undefined) {
```

Comment:
You can replace these types by the `StarkDatePickerFilter ` type you just created ;)